### PR TITLE
GC_fullCollect on sync threadpool

### PR DESCRIPTION
**Pull request by yglukhov**

### CODE REVIEW

File: lib/pure/concurrency/threadpool.nim [hunk 1]
```diff
@@ -526,10 +526,12 @@ proc nimSpawn4(fn: WorkerProc; data: pointer; id: ThreadId) {.compilerProc.} =
     if selectWorker(addr(distinguishedData[id]), fn, data): break
     await(distinguishedData[id].readyForTask)
 
+template spawnInAllThreads*(e: untyped) =
```
Comment (Araq) [inline lib/pure/concurrency/threadpool.nim]: I don't think this should be exported for now.

**Comment by yglukhov** [comment]
Done, also removed the extra wait after GC_fullCollect because it's unlikely user will want to wait for that.

**Comment by Araq** [comment]
Instead of GC_fullCollect you could nuke the whole thread local heap. API call: ``deallocOsPages()``

**Comment by yglukhov** [comment]
@Araq, will threadpool be still usable after `deallocOsPages`? Also what about finalizers? I think it's better to let them run.

**Comment by Araq** [comment]
Meh ok.

**Pull request merged**